rlel/App.xaml.cs

```diff
@@ -10,5 +11,47 @@ namespace rlel {
     /// Interaction logic for App.xaml
     /// </summary>
     public partial class App : Application {
+
+        /// <summary>The unique mutex name.</summary>
+        private const string UniqueMutexName = "{2E0604DE-AF17-4FD9-89F9-5762BD2030E8}";
+        private SingleInstance singleInstance;
+
+        /// <summary>The app on startup.</summary>
+        /// <param name="sender">The sender.</param>
+        /// <param name="e">The e.</param>
+        private void AppOnStartup (object sender, StartupEventArgs e) {
+            if (singleInstance == null) {
+                Guid guid = new Guid(UniqueMutexName);
+                singleInstance = new SingleInstance(guid);
```

Comment:
this is definitely not the correct usage of a mutex. you're checking the mutex state and then locking it, which defeats the purpose of having one. the mutex should be [static](http://stackoverflow.com/a/522874/385891)